### ISSUE TITLE
Add an option to unconstrain_draws to return posterior draws object

### ISSUE THREAD

**Issue by avehtari** [stan-dev/cmdstanr#873]
Currently, `unconstrain_draws()` accepts posterior draws object as an argument, but it always returns- a list of lists which has the contents in the different order than draws_list object. It would be nice to have an option to be able to get a proper draws object as output.

Example: I wanted to estimate variance from Pathfinder draws, and I ended up with the following doing the permutation of dimensions along the way, which is quite awful
```
summarise_draws(aperm(as_draws_array(fit1$unconstrain_draws(draws=as_draws(pth1))), c(3,2,1)), variance)
```

**Comment by jgabry** [comment]
@andrjohns Do you by any chance have time to look into this?

**Comment by andrjohns** [comment]
@jgabry good question! It's a tricky thing to implement, and a duplicate of #730, which I've been meaning to tackle for a while. I've got a couple of long plane trips tomorrow, so I'll have a bash at this to pass the time

**Comment by avehtari** [comment]
Not a duplicate, more like a simpler subset, and the solution for that subset is the onliner given in the first comment of this issue. Jut add `format` option to `unconstrain_draws` and `aperm` the dimensions. #730 is a more complicated issue.

**Comment by andrjohns** [comment]
> Not a duplicate, more like a simpler subset, and the solution for that subset is the onliner given in the first comment of this issue. Jut add `format` option to `unconstrain_draws` and `aperm` the dimensions. #730 is a more complicated issue.

Ah I see, well that's a bit quicker!

**Comment by andrjohns** [comment]
Closed by #886
